src/visitors/components/QrScanner.js

```diff
@@ -0,0 +1,99 @@
+/* global Instascan */
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+
+const Video = styled.video`
+  background-color: black;
+  border-radius: 0.2em;
+  width: 100%;
+`;
+
+
+const instascanAvailable = () => {
+  try {
+    return Boolean(Instascan.Scanner);
+  } catch (error) {
+    return false;
+  }
+};
+
+
+export default class QrScanner extends React.Component {
```

Comment:
Can this refactoring be pulled out into a separate PR?

Reply:
See #664